### PR TITLE
Feature: new and improved step detection and step length estimation

### PR DESCRIPTION
**Pull request by Dbraae**
### Feature Description
We implemented our own step detector using only accelerometer values (no step counter or detector anymore).
Furthermore, we changed which accelerometer values were used for step length estimation, which drastically improved the accuracy of the estimations.

This PR solves:
- [Our own implementation of Step Detection](https://trello.com/c/KSxLQBpE)
- [Accurate Scarlet step length estimation](https://trello.com/c/YYhqNyEn)
- [Accurate Weinberg step length estimation](https://trello.com/c/SiUeR2jH)

### Focus of Review
1. Your opinion on whether or not the added code is of high enough quality and understandable in terms of comments, naming and structure.
2. Check if the application detects your steps (they are detected as peaks).

### Files To Review
DeadReckoningService.kt

### How To Run
- Internet is required
- Add a valid API key to google_maps_api.xml
- Open Project with Android Studio
- Requires a Android Phone connected (and with developer mode enabled)
- Run as debug on Android
- Open the logcat in Android Studio
- Either shake your phone or take steps to see the logcat give you new peaks.* You can search for "Peak found" if the logcat is crowded.

*The step detector will count minor shaking of the phone as steps, ignore this for now, as we have not dealt with that issue.

### CODE REVIEW

File: DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt [hunk 1]
```diff
@@ -90,44 +91,34 @@ class DeadReckoningService : Service(), SensorEventListener {
...
-    private var stepCounterInitial: Int = 0
-    private var stepCounter: Int = 0
+    // Step detection and step length estimation fields
+    private val accArraySize: Int = 10000
 
-    // Step length estimation fields
-    private var accelerometerZReadings = mutableListOf<Pair<Long, Float>>()
+    private var accelerometerZReadings = arrayOfNulls<Pair<Float, Double>>(accArraySize)
+    private var accBufferIndex: Int = 0
+    private var gravitationalAccel: Double = 9.72
```
Comment (KVISDAOWNER) [inline DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt]: It is probably all good. I would just note that according to google the gravitational acceleration is 9.80665 m/s^2

**Review by Tutter — DISMISSED** [review]
Maybe add comments to the fields in line 95-101 of DeadReckoningService.kt to say what they're used for. Fine either way. Just a suggestion.

**Review by Tutter — DISMISSED** [review]

**Review by KVISDAOWNER — DISMISSED** [review]
I am able to run and get the expected results. Overall it looks very nice, good work! :D I don't understand much of what is done in the code, but I guess it makes more sense if coupled with the PDR doc.  I know that we do not have a well defined specification of how MVVM should be done, but doing this much calculation in a service, which is essentially a view, doesn't seem right. I am not saying that we should prioritize fixing this but we should be aware of it.

File: DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt (hunk 1 above)
Comment (Dbraae) [reply DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt]: I updated the value :)

**Review by KVISDAOWNER — APPROVED** [review]

**Review by Tutter — APPROVED** [review]

**Pull request merged**